enhancements/cloud-integration/tokenized-auth-enablement-operators-on-cloud.md

```diff
@@ -0,0 +1,340 @@
+for HyperShift)
+
+Bring Your Own Credentials (BYOC) where an operator can manage and distribute credentials to its operands are out of
+scope.
+
+
+## Proposal
+
+**Cloud Credential Operator (CCO) changes**: Adds a token-aware mode, while nominally set in "Manual" mode, CCO will
+look for and process CredentialRequests referenced in Operator CRs when:
+
+```bash
+# infrastructure platform is AWS
+$ oc get infrastructure cluster -o jsonpath --template '{ .status.platform }'
+AWS
+
+# credentialsMode is "Manual"
+$ oc get cloudcredential cluster -o jsonpath --template '{ .spec.credentialsMode }'
+Manual
+
+# serviceAccountIssuer is non empty
+$ oc get authentication cluster -o jsonpath --template='{ .spec.serviceAccountIssuer }'
+abutcher-oidc.s3.us-east-1.amazonaws.com
+```
+
+This will operate in one of two ways depending on which other parts of this EP for other components are adopted: 
+specifically HyperShift/Rosa adding CCO with these token-aware changes or continuing with the current support for the 
+pod identity webhook.
+
+CCO's token-aware mode will work by adding, for example for STS, a "role-arn"-type field on operator added 
+CredentialsRequest objects. This is a new API field on the CredentialsRequest spec. When CCO acquires a 
+CredentialsRequest it will create a Secret (as today mostly) with new functionality: Adding to the Secret a path to the
+projected ServiceAccount token (static for OCP) and a roleARN (from the new field on the CredentialsRequest).
+
+Validation of CredentialsRequest by this new token-aware CCO? Maybe. Currently, CCO gets some permissions under ROSA
+(unused), could expand these permissions to include tools like AWS's Policy Simulator such that it could validate a role
+has permissions, and with this CCO could be the alerting mechanism for a changed CredentialsRequest without sufficient 
+permissions.
+
+**HyperShift changes**: Include Cloud Credential Operator with token-aware mode (see above). Allows for processing of 
+CredentialsRequest objects added by operators.
+
+**OperatorHub and Console changes**: Allow for input from user of additional fields during install depending on the 
+Cloud Provider which will result in ENV variables on the Subscription object that are required in the 
+CredentialsRequests created by the Operator. Setting the Subscription config.ENV will allow UX for the information 
+needed by CCO or webhook, for input of the cloud credentials while not having to change the Subscription API.
+
+Show in OperatorHub that the cluster is in a mode that supports token-based authentication by reading the 
+`.spec.serviceAccountIssuer` from the Authentication CR, `.spec.platformSpec.type` from the Infrastructure CR,
+`.spec.credentialsMode` from the CloudCredentials CR.
+
+Show that the operator is enabled for Token-based use by reading the CSV annotation provided by the operator author.
```

Comment:
I think this is because that is how traditionally things have always been done in OLM/CSV side of things. An example that comes to my mind is - default namespace of the operator etc.

Reply:
Hemant states the OLM/CSV tradition correctly. TY. I think it fits in okay with that.

Reply:
I am though concerned how it will work when the operator is being installed via CLI. I can see this easily doable via console-gui, but how will users provide this value when they create subscriptions manually?

Reply:
Ok, so, the clusterserviceversion tells us how the operator could/should be deployed and tells us about the capability of the operator? Seems like STS support is a capability that could be first class, curious as to why annotations have been chosen in the past for this kind of thing?

Reply:
I wasn't part of the decision making on that, but i'd speculate:

ease/flexibility of introducing new ones w/o needing api updates to allow them to be specified, and the fact that some of these things are more openshift/downstream concepts that we wouldn't want/be able to encode into an upstream api

Reply:
yes, what Ben said were all the consideration on the not first-class part of the spec and just as annotations for now.